src/main/java/com/microsoft/azure/springcloudplayground/service/ServiceModule.java

```diff
@@ -2,43 +2,53 @@
 
 import com.microsoft.azure.springcloudplayground.metadata.Describable;
 import com.microsoft.azure.springcloudplayground.metadata.MetadataElement;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.NonNull;
 import lombok.Setter;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 @NoArgsConstructor
 public class ServiceModule extends MetadataElement implements Describable {
 
     @Setter
     private String description;
 
+    @Getter
+    private String portId;
```

Comment:
port not portId

Reply:
It can be somewhat like `Id` for web template to identify each input number. Use port there may be confusing.